### PR TITLE
Change my contact details changes

### PR DESCRIPTION
**Pull request by akssinhaamit**
# Motivation and Context
New feature account details changes

# What has changed
A new feature has been added

# How to test?

# Links

# Screenshots (if appropriate):

### CODE REVIEW

File: frontstage/views/account/account.py [hunk 1]
```diff
@@ -1,20 +1,76 @@
...
+
+@account_bp.route('/change-account-details', methods=['GET', 'POST'])
+@jwt_authorization(request)
+def change_account_details(session):
+    form = ContactDetailsChangeForm(request.values)
+    party_id = session.get_party_id()
+    respondent_details = party_controller.get_respondent_party_by_id(party_id)
+    update_required_flag = False
+    if request.method == 'POST' and form.validate():
+        logger.info('Attempting to create account')
```
Comment (insacuri) [inline frontstage/views/account/account.py:42]: This needs changing, we're changing the details not creating a new one

File: frontstage/views/account/account.py [hunk 2]
```diff
@@ -1,20 +1,76 @@
...
+        message = "We've updated your"
```
Comment (insacuri) [inline frontstage/views/account/account.py:43]: This whole bit feels like it should be in its own function with unit tests

File: frontstage/views/account/account.py [hunk 3]
```diff
@@ -1,20 +1,76 @@
...
+        if not form_valid:
+            flash('At least one option should be selected!')
+            return redirect(url_for('account_bp.get_account'))
+        if form.data['option'] == 'contact_details':
+            return redirect(url_for('account_bp.change_account_details'))
+    else:
+        return render_template('account/account.html', form=form, respondent=respondent_details)
 
+
+@account_bp.route('/change-account-details', methods=['GET', 'POST'])
```
Comment (insacuri) [inline frontstage/views/account/account.py:34]: I think the GET and POST should be in 2 separate functions to make them easier to understand

File: frontstage/views/account/account.py [hunk 4]
```diff
@@ -1,20 +1,76 @@
...
+            except ApiError as exc:
+                logger.error('Failed to updated account', status=exc.status_code)
+                raise exc
+            logger.info('Successfully updated account')
+            success_panel = message
+            return redirect(url_for('surveys_bp.get_survey_list', tag='todo', success_panel=success_panel))
+        else:
+            return redirect(url_for('surveys_bp.get_survey_list', tag='todo'))
+    else:
+        return render_template('account/account.contact-detail-change.html',
```
Comment (insacuri) [inline frontstage/views/account/account.py:75]: What is that . doing in there?  The file needs to be renamed to `account-contact-detail-change` and that name change reflected here

File: frontstage/views/account/account.py [hunk 5]
```diff
@@ -1,20 +1,76 @@
...
+                message += ", phone number"
+
+        if update_required_flag:
+            try:
+                party_controller.update_account(respondent_details)
+            except ApiError as exc:
+                logger.error('Failed to updated account', status=exc.status_code)
+                raise exc
+            logger.info('Successfully updated account')
+            success_panel = message
```
Comment (insacuri) [inline frontstage/views/account/account.py:70]: Could we not just `flash(message)`?  I think it'd be cleaner in the code, but this is pretty minor as these informational messages are done in so many ways.

File: frontstage/controllers/party_controller.py [hunk 6]
```diff
@@ -88,6 +88,21 @@ def create_account(registration_data):
     logger.info('Successfully created account')
 
 
+def update_account(respondent_data):
```
Comment (insacuri) [inline frontstage/controllers/party_controller.py:91]: There should be a simple unit test for this function.  Doesn't need to be much, but what does it return when you get a 200 or 404?  Looking at it, it just returns text so I think thats useful to highlight via a unit test

**Review by insacuri — CHANGES_REQUESTED** [review]
Sorry for all the comments >_<

File: frontstage/templates/account/account.contact-detail-change.html [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+        {{
+            onsButton({
+            "text": "Save",
+            "id": "btn-option-save",
+            "classes": 'btn-group__btn',
+            "submitType": "timer"
+            })
+        }}
+        <a href="{{ url_for('account_bp.get_account', tag='todo') }}"
+      role="button" class="btn btn--secondary" id="btn-upload-ci-cancel"><span class="btn__inner">Cancel</span></a>
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply frontstage/templates/account/account.contact-detail-change.html:144]: 👍

[the remark this answers is not in the document]
Comment (akssinhaamit) [reply frontstage/templates/account/account.contact-detail-change.html:144]: :+1

File: frontstage/templates/account/account.html [hunk 8]
```diff
@@ -26,5 +40,89 @@ <h1 class="u-fs-xl">Account details</h1>
...
+                "radios": [
+                    {
+                        "id": "contact_details",
+                        "label": {
+                            "text": "Change my contact details"
+                        },
+                        "value": "contact_details"
+                    },
+                    {
+                        "id": "business_details",
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply frontstage/templates/account/account.html:60]: 👍

File: frontstage/views/account/account.py (hunk 1 above)
Comment (akssinhaamit) [reply frontstage/views/account/account.py:42]: 👍

File: frontstage/views/account/account.py [hunk 9]
```diff
@@ -1,20 +1,76 @@
...
 @jwt_authorization(request)
 def get_account(session):
+    form = OptionsForm()
+    form_valid = form.validate()
     party_id = session.get_party_id()
     respondent_details = party_controller.get_respondent_party_by_id(party_id)
-    return render_template('account/account.html', respondent=respondent_details)
+    if request.method == 'POST':
+        if not form_valid:
+            flash('At least one option should be selected!')
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply frontstage/views/account/account.py:26]: 👍

File: frontstage/views/account/account.py (hunk 4 above)
Comment (akssinhaamit) [reply frontstage/views/account/account.py:75]: 👍

File: frontstage/views/account/account.py (hunk 2 above)
Comment (akssinhaamit) [reply frontstage/views/account/account.py:43]: ok

File: frontstage/views/account/account.py (hunk 3 above)
Comment (akssinhaamit) [reply frontstage/views/account/account.py:34]: ok

File: frontstage/views/account/account.py (hunk 5 above)
Comment (akssinhaamit) [reply frontstage/views/account/account.py:70]: ok

File: frontstage/controllers/party_controller.py (hunk 6 above)
Comment (akssinhaamit) [reply frontstage/controllers/party_controller.py:91]: its now a part of the integration test which suggests that.

**Review by insacuri — CHANGES_REQUESTED** [review]
A few more changes sorry

File: frontstage/views/account/account.py [hunk 10]
```diff
@@ -14,7 +17,75 @@
...
+    return render_template('account/account.html', form=form, respondent=respondent_details, error_panel=error_panel)
 
+
+@account_bp.route('/', methods=['POST'])
+@jwt_authorization(request)
+def update_account(session):
+    form = OptionsForm()
+    form_valid = form.validate()
+    if not form_valid:
+        error_panel = 'At least one option should be selected.'
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply frontstage/views/account/account.py:33]: done

File: frontstage/views/account/account.py [hunk 11]
```diff
@@ -14,7 +17,75 @@
...
+@account_bp.route('/change-account-details', methods=['GET', 'POST'])
+@jwt_authorization(request)
+def change_account_details(session):
+    form = ContactDetailsChangeForm(request.values)
+    party_id = session.get_party_id()
+    respondent_details = party_controller.get_respondent_party_by_id(party_id)
+    update_required_flag = False
+    attributes_changed = []
+    if request.method == 'POST' and form.validate():
+        logger.info('Attempting to update contact details changes on the account')
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply frontstage/views/account/account.py:48]: done

File: frontstage/views/account/account.py [hunk 12]
```diff
@@ -14,7 +17,75 @@
...
+            success_panel = create_success_message(attributes_changed, "We've updated your ")
+            return redirect(url_for('surveys_bp.get_survey_list', tag='todo', success_panel=success_panel))
+        else:
+            return redirect(url_for('surveys_bp.get_survey_list', tag='todo'))
+    else:
+        return render_template('account/account-contact-detail-change.html',
+                               form=form, errors=form.errors, respondent=respondent_details)
+
+
+def check_attribute_change(attributes_changed, respondent_details, update_required_flag):
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply frontstage/views/account/account.py:66]: done

File: frontstage/views/account/account.py [hunk 13]
```diff
@@ -14,7 +17,75 @@
...
+        update_required_flag = True
+        attributes_changed.append('last name')
+    if request.form.get('phone_number') != respondent_details['telephone']:
+        respondent_details['telephone'] = request.form.get('phone_number')
+        update_required_flag = True
+        attributes_changed.append('telephone number')
+    return update_required_flag
+
+
+def create_success_message(attr, message):
```
[the remark this answers is not in the document]
Comment (akssinhaamit) [reply frontstage/views/account/account.py:82]: done

**Review by insacuri — APPROVED** [review]
I've not tested it but it looks good to go.  Make sure to test it before merging and/or in preprod!

**Pull request merged**